File: backend/entity/keygame.go
// entity/key_game.go
package entity

import (
	"time"

	"gorm.io/gorm"
)

type KeyGame struct {
	gorm.Model

	GameID uint  `json:"game_id"`
	Game   *Game `gorm:"foreignKey:GameID" json:"game,omitempty"`

	KeyCode string `json:"key_code" gorm:"uniqueIndex"`

	// จองคีย์ให้ OrderItem ไหน (nil = ว่าง)
	OwnedByOrderItemID *uint      `json:"owned_by_order_item_id"`
	OwnedByOrderItem   *OrderItem `gorm:"foreignKey:OwnedByOrderItemID" json:"owned_by_order_item,omitempty"`

	// สถานะการเปิดเผยคีย์
	IsRevealed bool       `json:"is_revealed" gorm:"default:false"`
	RevealedAt *time.Time `json:"revealed_at"`
}

File: frontend/src/interfaces/KeyGame.ts
// src/interfaces/KeyGame.ts
import type { Game } from "./Game";
import type { OrderItem } from "./OrderItem";

export interface KeyGame {
  ID: number;
  key: string;
  game_id: number;
  game?: Game;
  order_item_id?: number | null;
  order_item?: OrderItem;
}

export interface CreateKeyGameRequest {
  key?: string;
  game_id: number;
}
